controllers/configurationpolicy_controller.go

```diff
@@ -1538,8 +1538,19 @@ func (r *ConfigurationPolicyReconciler) handleSingleObj(
 			// it is a must have and it does exist, so it is compliant
 			compliant = true
 			if strings.EqualFold(string(remediation), string(policyv1.Enforce)) {
-				statusUpdateNeeded = createStatus("", obj.gvr.Resource, compliantObject, obj.namespaced, obj.policy,
-					obj.index, compliant, true)
+				if updatedObj {
+					// object updated in checkAndUpdateResource, send event
+
+					log.Info("Updated must have object", "resource", obj.gvr.Resource, "name", obj.name)
+					reason := "K8s update success"
+					idStr := identifierStr([]string{obj.name}, obj.namespace, obj.namespaced)
+					msg := fmt.Sprintf("%v %v was updated successfully", obj.gvr.Resource, idStr)
+
+					statusUpdateNeeded = addConditionToStatus(obj.policy, obj.index, true, reason, msg)
```

Comment:
I'm wondering if this should remain compliant here since that is different behavior than when the object is created by a musthave ConfigurationPolicy.

The switching from compliant to noncompliant can be a signal for a user to investigate why it had changed.

Reply:
I'm confused - this is compliant, right? I have compliant set to true and when I tested it the status was compliant

Reply:
@willkutler right, I was questioning if it actually should be compliant like you have it set to. I'm leaning towards making it noncompliant. Let's have a team Slack discussion on it.

Reply:
Oh, that makes sense. Personally I would rather have the creation event be compliant and keep the update compliant, since those events aren't really violations. I'll make a post in the team slack channel about it.